File: Templates/Java/integrationtesting/src/main/java/com/example/integrationtesting/UserServiceImpl.java
package com.example.integrationtesting;

import java.util.List;
import java.util.Optional;

import org.springframework.stereotype.Service;

import tools.jackson.core.type.TypeReference;
import tools.jackson.databind.MapperFeature;
import tools.jackson.databind.json.JsonMapper;

@Service
public class UserServiceImpl implements UserService {
    private final JsonPlaceholderClient client;
    private final JsonMapper mapper = JsonMapper
        .builder()
        .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true)
        .configure(MapperFeature.ALLOW_COERCION_OF_SCALARS, false)
        .build();

    public UserServiceImpl(JsonPlaceholderClient client) {
        this.client = client;
    }

    public List<User> getAll() {
        var string = client.get("/users", Optional.empty());
        var json = mapper.readTree(string);
        return mapper.convertValue(json, new TypeReference<List<User>>() {});
    }

    public Optional<User> getById(String id) {
        var string = client.get("/users", Optional.empty());
        var json = mapper.readTree(string);
        var allUsers = mapper.convertValue(json, new TypeReference<List<User>>() {});
        var user = allUsers.stream().filter(u -> id.equals(u.id())).findAny();
        return user;
    }

    public List<AddressUser> getNearby(double lat, double lng, double miles) {
        var string = client.get("/users", Optional.empty());
        var json = mapper.readTree(string);
        var allUsers = mapper.convertValue(json, new TypeReference<List<AddressUser>>() {});
        return allUsers.stream().filter(user ->
            user.address().geo().isPresent() &&
            DistanceUtils.calculateDistance(lat, lng, Double.parseDouble(user.address().geo().get().lat()), Double.parseDouble(user.address().geo().get().lng())) <= miles
        ).toList();
    }
}

File: Templates/Kotlin/integrationtesting/src/main/java/com/example/integrationtesting/UserServiceImpl.kt
package com.example.integrationtesting

import com.example.integrationtesting.DistanceUtils.calculateDistance
import org.springframework.stereotype.Service
import tools.jackson.core.type.TypeReference
import tools.jackson.databind.MapperFeature
import tools.jackson.databind.json.JsonMapper
import kotlin.streams.asSequence;

@Service
internal class UserServiceImpl(private val client: JsonPlaceholderClient) : UserService {
    private val mapper: JsonMapper = JsonMapper
        .builder()
        .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true)
        .configure(MapperFeature.ALLOW_COERCION_OF_SCALARS, false)
        .build()

    override val all: MutableList<User?>?
        get() {
            val string = client.get("/users", null)
            val json = mapper.readTree(string)
            return mapper.convertValue<MutableList<User?>?>(
                json,
                object :
                    TypeReference<MutableList<User?>?>() {})
        }

    override fun getById(id: String): User? {
        val string = client.get("/users", null)
        val json = mapper.readTree(string)
        val allUsers = mapper.convertValue<MutableList<User?>>(json, object : TypeReference<MutableList<User?>?>() {})
        val user: User? = allUsers.stream().asSequence().filter { u: User? -> id == u?.id.toString() }.firstOrNull();
        return user
    }

    override fun getNearby(lat: Double, lng: Double, miles: Double): MutableList<AddressUser?>? {
        val string = client.get("/users", null)
        val json = mapper.readTree(string)
        val allUsers = mapper.convertValue<MutableList<AddressUser?>>(
            json,
            object : TypeReference<MutableList<AddressUser?>?>() {})
        return allUsers.stream().filter { user: AddressUser? ->
            user!!.address!!.geo!!.isPresent() &&
                    calculateDistance(
                        lat,
                        lng,
                        user.address.geo.get().lat,
                        user.address.geo.get().lng
                    ) <= miles
        }.toList()
    }
}